exchanges/multipart-fetch/src/multipartFetchExchange.ts

```diff
@@ -0,0 +1,194 @@
+import { extractFiles } from 'extract-files';
+import {
+  Exchange,
+  Operation,
+  OperationResult,
+  makeResult,
+  makeErrorResult,
+} from '@urql/core';
+
+interface Body {
+  query: string;
+  variables: void | object;
+  operationName?: string;
+}
+
+const isOperationFetchable = (operation: Operation) =>
+  operation.operationName === 'query' || operation.operationName === 'mutation';
+
+export const multipartFetchExchange: Exchange = ({ forward }) => ops$ => {
+  const sharedOps$ = share(ops$);
+
+  const fetchResults$ = pipe(
+    sharedOps$,
+    filter(isOperationFetchable),
+    mergeMap(operation => {
+      const teardown$ = pipe(
+        sharedOps$,
+        filter(
+          op => op.operationName === 'teardown' && op.key === operation.key
+        )
+      );
+
+      return pipe(
+        createFetchSource(
+          operation,
+          operation.operationName === 'query' &&
+            !!operation.context.preferGetMethod
+        ),
+        takeUntil(teardown$)
+      );
+    })
+  );
+
+  const forward$ = pipe(
+    sharedOps$,
+    filter(op => !isOperationFetchable(op)),
+    forward
+  );
+
+  return merge([fetchResults$, forward$]);
+};
+
+const getOperationName = (query: DocumentNode): string | null => {
+  const node = query.definitions.find(
+    (node: any): node is OperationDefinitionNode => {
+      return node.kind === Kind.OPERATION_DEFINITION && node.name;
+    }
+  );
+
+  return node && node.name ? node.name.value : null;
+};
+
+const createFetchSource = (operation: Operation, shouldUseGet: boolean) => {
+  if (
+    process.env.NODE_ENV !== 'production' &&
+    operation.operationName === 'subscription'
+  ) {
+    throw new Error(
+      'Received a subscription operation in the httpExchange. You are probably trying to create a subscription. Have you added a subscriptionExchange?'
+    );
+  }
+
+  return make<OperationResult>(({ next, complete }) => {
+    const abortController =
+      typeof AbortController !== 'undefined'
+        ? new AbortController()
+        : undefined;
+
+    const { context } = operation;
+    const { files } = extractFiles(operation.variables);
+
+    const extraOptions =
+      typeof context.fetchOptions === 'function'
+        ? context.fetchOptions()
+        : context.fetchOptions || {};
+
+    const operationName = getOperationName(operation.query);
+
+    const body: Body = {
+      query: print(operation.query),
+      variables: operation.variables,
+    };
+
+    if (operationName !== null) {
+      body.operationName = operationName;
+    }
+
+    const fetchOptions = {
+      ...extraOptions,
+      method: shouldUseGet ? 'GET' : 'POST',
+      headers: {
+        'content-type': 'application/json',
+        ...extraOptions.headers,
+      },
+      signal:
+        abortController !== undefined ? abortController.signal : undefined,
+    };
+
+    if (!!files.size) {
+      fetchOptions.body = new FormData();
```

Comment:
Is there anything specific we need to do to support this in Node.js? Can we check what `isomorphic-unfetch` / `node-fetch` requires for this to work correctly?

Reply:
Both show this working with `.body = FormData`

Reply:
Awesome!